src/components/PopupWithDelete.js

```diff
@@ -0,0 +1,25 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithDelete extends Popup {
+  constructor(popupSelector, handleDeleteConfirm) {
+    super(popupSelector);
+    this._handleDeleteConfirm = null;
+  }
+
+  setHandleConfirm(handleConfirm) {
+    this._handleDeleteConfirm = handleConfirm;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+
+    // Add an event listener for the delete confirmation button
+    this._confirmButton = this._popupElement.querySelector(
+      ".modal__button_confirm"
+    );
+    this._confirmButton.addEventListener("click", () => {
+      this._handleDeleteConfirm();
+      this.close();
```

Comment:
Please, delete `this.close();`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**

Reply:
I hope I understand right here. and I also deleted in the popupwithform.